File: sentiments.py
import sys
from googletrans import Translator
from textblob import TextBlob

# Recibimos el texto desde PHP
texto_usuario = ' '.join(sys.argv[1:])  # Unimos todos los argumentos en una sola cadena

try:
    # Creamos un objeto Translator de Google
    translator = Translator()

    # Traducimos el texto al inglés
    translated_text = translator.translate(texto_usuario, dest='en').text

    # Creamos un nuevo objeto TextBlob con el texto traducido
    translated_blob = TextBlob(translated_text)

    # Obtenemos el sentimiento del texto traducido
    sentimiento = translated_blob.sentiment

    # Convertimos el sentimiento detectado en un estado de ánimo
    e_animo = ""
    print("Polaridad del sentimiento:", sentimiento.polarity)

    if sentimiento.polarity > 0.3:
        e_animo = "feliz"
    elif sentimiento.polarity < -0.3:
        e_animo = "triste"
    elif 0 <= sentimiento.polarity <= 0.3:
        e_animo = "neutro"
    elif 0.3 < sentimiento.polarity <= 0.5:
        e_animo = "contento"
    elif 0.5 < sentimiento.polarity <= 0.7:
        e_animo = "emocionado"
    elif 0.7 < sentimiento.polarity <= 0.9:
        e_animo = "entusiasmado"
    else:
        e_animo = "ansioso"

    # Devolvemos el estado de ánimo al script PHP
    print(e_animo)

except Exception as e:
    print("Error:", e)

File: sentiments.php
<?php

require 'vendor/autoload.php';
use GuzzleHttp\Client;

// Definimos el texto de entrada del usuario
$texto_usuario = "Estoy super aburrido";

$e_animo = exec("python sentiments.py \"" . $texto_usuario . "\"");





// Definimos los géneros deseados
$genre_ids_array = [];

switch ($e_animo) {
    case "feliz":
        $genre_ids_array = [28, 12, 16, 35];
        break;
    case "triste":
        $genre_ids_array = [10752, 18, 80, 10751];
        break;    
    case "neutro":
        $genre_ids_array = [878, 12, 28, 27];
        break;
    case "contento":
        $genre_ids_array = [28, 12, 878];
        break;
    case "emocionado":
        $genre_ids_array = [18, 27, 53];
        break;
    case "entusiasmado":
        $genre_ids_array = [35, 10751, 10749];
        break;
    case "ansioso":
        $genre_ids_array = [53, 9648, 10749];
        break;
    default:
        echo "Estado de ánimo no reconocido";
        exit;
}

// Definimos la clave de la API de TMDb
$api_key = 'b2d11b0c7ab0e17b36609db82291a79f'; // Reemplaza con tu API Key de TMDb

// Configuramos el cliente Guzzle HTTP
$client = new Client([
    'base_uri' => 'https://api.themoviedb.org/3/',
]);

try {
    // Realizamos la llamada a la API de TMDb para obtener películas de los géneros deseados
    $response = $client->request('GET', 'discover/movie', [
        'query' => [
            'api_key' => $api_key,
            'with_genres' => implode(',', $genre_ids_array), // Especificamos los géneros deseados
            'language' => 'es', // Establecemos el idioma de las películas (opcional)
            'page' => 1 // Página de resultados (opcional)
        ]
    ]);

    // Decodificamos la respuesta JSON
    $data = json_decode($response->getBody(), true);

    // Mostramos las películas de los géneros deseados
    echo "Películas adecuadas para cuando estás " . $e_animo . ":\n";
    foreach ($data['results'] as $movie) {
        echo $movie['title'] . " (" . $movie['release_date'] . ")\n";
        echo "<img src='https://image.tmdb.org/t/p/w500" . $movie['poster_path'] . "' alt='Póster de " . $movie['title'] . "'><br>";
        echo "<img src='https://image.tmdb.org/t/p/original" . $movie['backdrop_path'] . "' alt='Fondo de " . $movie['title'] . "'><br>";
    }
} catch (Exception $e) {
    // Manejamos errores de la API
    echo 'Error al obtener películas: ' . $e->getMessage();
}

?>
